app.js

```diff
@@ -121,4 +121,67 @@ form.addEventListener('submit', (event) => {
     const error = document.getElementById('error-message');
     error.style.display = 'block';
   }
-});
\ No newline at end of file
+});
+
+// Implement the local storage logic to store inputs data.
+// function that detects whether localStorage is both supported and available,I took it from MDN.
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+      // everything except Firefox
+      e.code === 22
+      // Firefox
+      || e.code === 1014
+      // test name field too, because code might not be present
+      // everything except Firefox
+      || e.name === 'QuotaExceededError'
+      // Firefox
+      || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+      // acknowledge QuotaExceededError only if there's something already stored
+      && (storage && storage.length !== 0);
+  }
+}
+// Getting values from storage
+function setInputsData() {
+  const inputsDataObj = JSON.parse(localStorage.getItem('inputsData'));
+  document.getElementById('fullname').value = inputsDataObj.name;
+  document.getElementById('email').value = inputsDataObj.email;
+  document.getElementById('message').value = inputsDataObj.message;
+}
+
+// Setting values in storage
+function populateStorage() {
+  const inputsDataObj = {};
+
+  inputsDataObj.name = document.getElementById('fullname').value;
+  inputsDataObj.email = document.getElementById('email').value;
+  inputsDataObj.message = document.getElementById('message').value;
+  localStorage.setItem('inputsData', JSON.stringify(inputsDataObj));
+  setInputsData();
+}
+
+if (storageAvailable('localStorage')) {
+  // Yippee! We can use localStorage awesomeness
+  // Testing whether your storage has been populated
+  if (!localStorage.getItem('inputsData')) {
+    populateStorage();
+  } else {
+    setInputsData();
+  }
+}
+
```

Comment:
- [ ] Kindly make sure to write code for `when the user loads the page, any saved data in the local storage have input fields with pre-filled data`. 
I have attached a video where when we refresh the page the data on the input files should be persistent. You have to use `window.onload ` for this. Kindly work on it.

https://user-images.githubusercontent.com/70269507/162326319-3d65e196-b835-40be-b5ed-7255b344d1c4.mov

Reply:
Hi @sja-thedudem, I hope you are doing well,
**I notice that the event listener I used for this activity (`''change"`)** works fine when the input loses focus, so I used also the **`window.onload`**  but it behaves just one the same manner, I think I should use the `"input"` event listener to **capture the changes instantly.**

Reply:
### Done with "input" event listener.